### PR TITLE
[Typing][A-25,A-65] Add type annotations for `paddle/nn/{layer,functional}/vision.py`

### PR DESCRIPTION
**Pull request by Asthestarsfalll**
### PR Category

User Experience

### PR Types

Improvements

### Description

Add type annotations for paddle/nn/{layer,functnal}/vision.py
- #65008

### CODE REVIEW

File: python/paddle/nn/functional/vision.py [hunk 1]
```diff
@@ -373,7 +391,12 @@ def pixel_shuffle(x, upscale_factor, data_format="NCHW", name=None):
         return out
 
 
-def pixel_unshuffle(x, downscale_factor, data_format="NCHW", name=None):
+def pixel_unshuffle(
+    x: Tensor,
+    downscale_factor: int,
+    data_format: DataLayout2D | str = 'NCHW',
```
Comment (megemini) [inline python/paddle/nn/functional/vision.py]: 需要 `| str` ？

File: python/paddle/nn/functional/vision.py [hunk 2]
```diff
@@ -436,7 +459,12 @@ def pixel_unshuffle(x, downscale_factor, data_format="NCHW", name=None):
     return out
 
 
-def channel_shuffle(x, groups, data_format="NCHW", name=None):
+def channel_shuffle(
+    x: Tensor,
+    groups: int,
+    data_format: DataLayout2D | str = 'NCHW',
```
Comment (megemini) [inline python/paddle/nn/functional/vision.py]: 同上 ～

File: python/paddle/nn/functional/vision.py (hunk 1 above)
Comment (Asthestarsfalll) [reply python/paddle/nn/functional/vision.py]: 忘记去掉了，已修改

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**